### PR TITLE
Updating Acceptance tests from linter

### PR DESCRIPTION
**Pull request by teddylear**
### Description

Updates from tfproviderlint results for Acceptance tests

### Acceptance tests
- Acceptance tests have been updated based on results from runnign tfproviderlint

### References
Addressing from this request. 
https://github.com/hashicorp/terraform-provider-kubernetes/issues/887

### CODE REVIEW

**Review by dak1n1 — COMMENTED** [review]
Thanks for taking this on! These are some good changes. I noticed a few files were the might be some inconsistent naming, so I left a comment there, but otherwise this looks really good.

**Comment by teddylear** [comment]
@dak1n1 I can make the updates suggested above, one issue I faced though when trying to merge master was the following

`NOTE` This was running build on the master branch. 
```
[I] ➜ make build
==> Checking that code complies with gofmt requirements...
go install
# github.com/moby/term
vendor/github.com/moby/term/tc.go:13:28: undefined: unix.SYS_IOCTL
vendor/github.com/moby/term/tc.go:18:28: undefined: unix.SYS_IOCTL
vendor/github.com/moby/term/termios_bsd.go:24:31: undefined: unix.SYS_IOCTL
vendor/github.com/moby/term/termios_bsd.go:37:31: undefined: unix.SYS_IOCTL
make: *** [build] Error 2
```
Are you having the same problem? Not sure if I'm missing something obvious as to why this would be failing.

File: kubernetes/resource_kubernetes_cluster_role_test.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 func TestAccKubernetesClusterRole_basic(t *testing.T) {
 	var conf api.ClusterRole
 	resourceName := "kubernetes_cluster_role.test"
-	name := acctest.RandomWithPrefix("tf-acc-test")
+	name := fmt.Sprintf("tf-acc-test:%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
```
[the remark this answers is not in the document]
Comment (teddylear) [reply kubernetes/resource_kubernetes_cluster_role_test.go]: There is no benefit, changed this to be use the helper lib instead.

File: kubernetes/resource_kubernetes_config_map_test.go [hunk 2]
```diff
@@ -172,17 +156,18 @@ func TestAccKubernetesConfigMap_binaryData(t *testing.T) {
...
+		IDRefreshName: resourceName,
 		Providers:     testAccProviders,
 		CheckDestroy:  testAccCheckKubernetesConfigMapDestroy,
 		Steps: []resource.TestStep{
 			{
 				Config: testAccKubernetesConfigMapConfig_generatedName(prefix),
 				Check: resource.ComposeAggregateTestCheckFunc(
-					testAccCheckKubernetesConfigMapExists("kubernetes_config_map.test", &conf),
+					testAccCheckKubernetesConfigMapExists(resourceName, &conf),
 					resource.TestCheckResourceAttr("kubernetes_config_map.test", "metadata.0.annotations.%", "0"),
```
[the remark this answers is not in the document]
Comment (teddylear) [reply kubernetes/resource_kubernetes_config_map_test.go]: Changed this to only use `resourceName`

**Comment by dak1n1** [comment]
Thanks, this looks good. I'm doing one final read-through and then waiting for tests to pass in our internal CI systems. The results of local testing look good. All makefile targets pass (except for testacc, which I'm running in CI instead).

```
$ make build depscheck fmtcheck test test-compile tests-lint tests-lint-fix tools vet website-lint website-lint-fix
==> Checking that code complies with gofmt requirements...
go install
==> Checking source code with 'git diff'...
==> Checking source code with go mod tidy...
==> Checking source code with go mod vendor...
make: Nothing to be done for 'fmtcheck'.
go test "/home/dakini/go/src/github.com/hashicorp/terraform-provider-kubernetes/kubernetes" || exit 1
ok      github.com/hashicorp/terraform-provider-kubernetes/kubernetes   0.114s
echo "/home/dakini/go/src/github.com/hashicorp/terraform-provider-kubernetes/kubernetes" | \
        xargs -t -n4 go test  -timeout=30s -parallel=4
go test '-timeout=30s' '-parallel=4' /home/dakini/go/src/github.com/hashicorp/terraform-provider-kubernetes/kubernetes
ok      github.com/hashicorp/terraform-provider-kubernetes/kubernetes   0.134s
go test -c "/home/dakini/go/src/github.com/hashicorp/terraform-provider-kubernetes/kubernetes"
go install github.com/bflad/tfproviderdocs
go install github.com/client9/misspell/cmd/misspell
go install github.com/katbyte/terrafmt
go mod tidy
go mod vendor
==> Checking acceptance test terraform blocks code with terrafmt...
==> Fixing acceptance test terraform blocks code with terrafmt...
make: Nothing to be done for 'tools'.
go vet .
==> Checking website against linters...
==> Running markdownlint-cli using DOCKER='podman --cgroup-manager=cgroupfs', DOCKER_RUN_OPTS='--userns=keep-id' and DOCKER_VOLUME_OPTS='rw,Z'
==> Running terrafmt diff...
==> Statically compiling provider for tfproviderdocs...
==> Getting provider schema for tfproviderdocs...

Initializing the backend...

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
==> Running tfproviderdocs...
==> Applying automatic website linter fixes...
==> Running markdownlint-cli --fix using DOCKER='podman --cgroup-manager=cgroupfs', DOCKER_RUN_OPTS='--userns=keep-id' and DOCKER_VOLUME_OPTS='rw,Z'
==> Fixing website terraform blocks code with terrafmt...
[dakini@dax terraform-provider-kubernetes ((401c3e3...) $)]$ git status
HEAD detached at teddylear/feature/ATLinterFixes
nothing to commit, working tree clean
```

(Internal links for my team to reference: https://ci-oss.hashicorp.engineering/viewLog.html?buildId=148147 https://ci-oss.hashicorp.engineering/viewLog.html?buildId=148148 )

**Review by dak1n1 — APPROVED** [review]
LGTM, no new test failures in CI.

**Pull request merged**